instrumentation/google.golang.org/grpc/otelgrpc/interceptor.go

```diff
@@ -368,7 +368,7 @@ func UnaryServerInterceptor(opts ...Option) grpc.UnaryServerInterceptor {
 		var statusCode grpc_codes.Code
 		defer func(t time.Time) {
 			elapsedTime := time.Since(t) / time.Millisecond
-			attr = append(attr, semconv.RPCGRPCStatusCodeKey.Int64(int64(statusCode)))
+			attr := append(metricAttrs, semconv.RPCGRPCStatusCodeKey.Int64(int64(statusCode)))
```

Comment:
Given `metricAttrs` and `attrs` share the same backing array, this will mutate `attrs` as well.

```suggestion
			attr := append([]attribute.KeyValue{semconv.RPCGRPCStatusCodeKey.Int64(int64(statusCode))}, metricAttrs...)
```

Reply:
This doesn't modify the underlying slice when I tested by forcing peerInfo to return data.  Either way, I switched to the suggestion.